packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+
+### Examples
+
+Given a data value of 2856 (bytes):
+
+```HTML
+<nimble-table>
+    <nimble-table-column-file-size field-name="size" unit-type="decimal">Size</nimble-table-column-file-size>
+</nimble-table>
+```
+
+![Decimal unit](../spec-images/fileSizeDecimal.png)
+
+```HTML
+<nimble-table>
+    <nimble-table-column-file-size field-name="size" unit-type="binary">Size</nimble-table-column-file-size>
+</nimble-table>
+```
+
+![Decimal unit](../spec-images/fileSizeBinary.png)
+
+### Anatomy
+
+### Angular integration
+
+An Angular directive will be created for the component. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component.
+
+### Visual Appearance
+
+Will use same text styling as the `nimble-table-column-text`.
+
+Values will be left-aligned. Right-alignment is useful when numeric values have aligned decimal points and are given in the same units. In that case, right-alignment allows for quick scanning to determine relative magnitude. But our column will not use a consistent number of decimal digits, nor a consistent unit. This leaves no clear motivation for right-aligning the values, while left-aligning provides better association with the column header. However, if we discover that there is value in right-aligning file sizes, we can easily make that change in a non-breaking way in the future.
+
+## Implementation
+
+Cell view will extend `TableColumnTextCellViewBase` and use its template.
+
+```ts
+class TableColumnFileSizeCellView extends TableColumnTextCellViewBase<
+    TableColumnFileSizeCellRecord,
+    TableColumnFileSizeColumnConfig
+> {
+    ...
+
+    private updateText(): void {
+        if (this.columnConfig) {
+            this.text = formatFileSize(
+                this.columnConfig.formatter, // instance of Intl.NumberFormat
+                this.unitStrings, // either ['byte', 'bytes', 'KB', ...] or ['byte', 'bytes', 'KiB', ...]
+                this.columnConfig.unitType,
+                this.cellRecord?.value
+            );
+        } else {
+            this.text = '';
+        }
+    }
+}
+
+// Shared function also used by group header view
+function formatFileSize(
+    formatter: Intl.NumberFormat,
+    unitStrings: string[],
+    unitType: FileSizeUnitType,
+    byteCount: number
+): string {
+    let currentSize = byteCount;
+    let unitIndex = 0;
+    if (currentSize !== 1) {
+        const divisor = unitType === FileSizeUnitType.binary ? 1024 : 1000;
+        while (currentSize >= divisor && unitIndex < unitStrings.length) {
+            currentSize /= divisor;
+            unitIndex += 1;
+        }
+    }
+
+    return `${formatter.format(currentSize)} ${unitStrings[unitIndex]}`;
+}
+```
+
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
```

Comment:
Are there any invalid states? What would we show for negative numbers?

Reply:
I don't think there are any. The only configuration option is `unit-type`, which will be strongly typed to disallow invalid values. For invalid values like `Inf`, we just show an empty cell rather than putting the whole column (and table) in an invalid state. I think we should support negative numbers in case a client wants to use this column to represent some kind of size delta.